### ISSUE TITLE
[Bug]: Tool use with `llama3-groq` without streaming

### ISSUE THREAD

**Issue by Clad3815** [BerriAI/litellm#4771]
### What happened?

This bug happen when using llama3 with groq with tool usage. This error not happen while using streaming mode. Here is the code to reproduce (Using OpenAI SDK)

```javascript
const OpenAI = require('openai');

// Initialize OpenAI client
const openai = new OpenAI({ apiKey: "sk-1234", baseURL: "http://localhost:4000/v1" });

// Use Streaming
const USE_STREAMING = true;
// const USE_STREAMING = false;

// Simple tool function
async function getCurrentTime() {
    return { time: new Date().toISOString() };
}

// Define the tool
const tools = [
    {
        type: 'function',
        function: {
            name: 'getCurrentTime',
            description: 'Get the current time',
            parameters: {
                type: 'object',
                properties: {},
            },
            function: getCurrentTime,
            parse: JSON.parse,
        },
    },
];

async function main() {
    const stream = await openai.beta.chat.completions.runTools({
        model: 'groq/llama3-groq-70b-8192-tool-use-preview',
        messages: [
            { role: 'system', content: 'Use your tools smartly' },
            { role: 'user', content: 'What time is it now? Use your tool' },
        ],
        tools: tools,
        stream: USE_STREAMING,
    })
        .on('functionCall', (functionCall) => console.log('functionCall', functionCall))
        .on('functionCallResult', (functionCallResult) => console.log('functionCallResult', functionCallResult))
    if (USE_STREAMING) {
        for await (const chunk of stream) {
            if (chunk.choices[0].delta.content) {
                console.log(chunk.choices[0].delta);
            }
        }
    }
    const completion = await stream.finalChatCompletion();
    console.log(JSON.stringify(completion, null, 2));
}

main().catch(console.error);
```

I'm using docker v1.41.24 with this config
```
  - model_name: groq/llama3-groq-70b-8192-tool-use-preview
    litellm_params:
      model: groq/llama3-groq-70b-8192-tool-use-preview
      api_base: https://api.groq.com/openai/v1
```

### Relevant log output

```shell
litellm-1  | Traceback (most recent call last):
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 2891, in chat_completion
litellm-1  |     responses = await llm_responses
litellm-1  |                 ^^^^^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 655, in acompletion
litellm-1  |     raise e
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 643, in acompletion
litellm-1  |     response = await self.async_function_with_fallbacks(**kwargs)
litellm-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 2487, in async_function_with_fallbacks
litellm-1  |     raise original_exception
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 2349, in async_function_with_fallbacks
litellm-1  |     raise e
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 2332, in async_function_with_fallbacks
litellm-1  |     response = await self.async_function_with_retries(*args, **kwargs)
litellm-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 2587, in async_function_with_retries
litellm-1  |     raise original_exception
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 2509, in async_function_with_retries
litellm-1  |     response = await original_function(*args, **kwargs)
litellm-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 788, in _acompletion
litellm-1  |     raise e
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/router.py", line 760, in _acompletion
litellm-1  |     response = await _response
litellm-1  |                ^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 1524, in wrapper_async
litellm-1  |     raise e
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 1336, in wrapper_async
litellm-1  |     result = await original_function(*args, **kwargs)
litellm-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/main.py", line 418, in acompletion
litellm-1  |     raise exception_type(
litellm-1  |           ^^^^^^^^^^^^^^^
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 8020, in exception_type
litellm-1  |     raise e
litellm-1  |   File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 6321, in exception_type
litellm-1  |     raise BadRequestError(
litellm-1  | litellm.exceptions.BadRequestError: litellm.BadRequestError: GroqException - 'messages.2' : for 'role:assistant' the following must be satisfied[('messages.2.function_call' : Value is not nullable)] LiteLLM Retried: 2 times, LiteLLM Max Retries: 3
```

**Comment by ghost** [comment]
hey @Clad3815 can you share your server logs - looking for the raw request sent by litellm

**Comment by ghost** [comment]
> litellm-1  | litellm.exceptions.BadRequestError: litellm.BadRequestError: GroqException - 'messages.2' : for 'role:assistant' the following must be satisfied[('messages.2.function_call' : Value is not nullable)] LiteLLM Retried: 2 times, LiteLLM Max Retries: 3


this does look like an error being raised by groq

**Comment by ghost** [comment]
Hey @Clad3815 just ran your script locally, this works for me 
<img width="1629" alt="Screenshot 2024-07-20 at 7 02 52 PM" src="https://github.com/user-attachments/assets/fab12b37-682f-4333-a31c-686308ed8884">

## Debug Server Logs: 
<img width="897" alt="Screenshot 2024-07-20 at 7 03 24 PM" src="https://github.com/user-attachments/assets/3c472bc3-7684-4fc4-9422-7c8b3e33461b">
<img width="1630" alt="Screenshot 2024-07-20 at 7 03 42 PM" src="https://github.com/user-attachments/assets/027d6d9d-9f90-45f6-8743-37bd864820f4">

**Comment by ghost** [comment]
@Clad3815 can you try v1.41.25 and if the issue persists, share your server logs as well?

**Comment by artur-pf** [comment]
Bug still persists for me, but is solved if version ` litellm==1.37.16` is used, instead of the newer ones. Please fix

**Comment by ghost** [comment]
@artur-pf unable to repro. This works for me on latest. Please create a new ticket with the error message + steps to repro. 
<img width="1585" alt="Screenshot 2024-07-28 at 1 23 24 PM" src="https://github.com/user-attachments/assets/e53cf80e-0e65-4d85-8a16-7c881191754d">

## My script

```js
const { OpenAI } = require('openai');

const openai = new OpenAI({
  apiKey: "sk-1234", // This is the default and can be omitted
  baseURL: "http://0.0.0.0:4000"
});

// async function main() {
//   const chatCompletion = await openai.chat.completions.create({
//     messages: [{ role: 'user', content: 'Say this is a test' }],
//     model: 'gpt-3.5-turbo',
//   }, {"headers": {
//     "Cache-Control": "s-maxage=0"
//   }});
// }

// async function main() {
//   const messages = [{"role": "user", "content": "What's the weather like in Boston today?"}];
//   const tools = [
//       {
//         "type": "function",
//         "function": {
//           "name": "get_current_weather",
//           "description": "Get the current weather in a given location",
//           "parameters": {
//             "type": "object",
//             "properties": {
//               "location": {
//                 "type": "string",
//                 "description": "The city and state, e.g. San Francisco, CA",
//               },
//               "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
//             },
//             "required": ["location"],
//           },
//         }
//       }
//   ];

//   const response = await openai.chat.completions.create({
//     model: "azure-gpt-3.5",
//     messages: messages,
//     tools: tools,
//     tool_choice: "auto",
//     stream: true
//   });

//   for await (const chunk of response) {
//     if (chunk.choices[0].delta.tool_calls != null) {
//       console.log(chunk.choices[0].delta.tool_calls[0]);
//     }
//   }
  
// }

// main();

// const OpenAI = require('openai');

// Initialize OpenAI client
// const openai = new OpenAI({ apiKey: "sk-1234", baseURL: "http://localhost:4000/v1" });

// Use Streaming
// const USE_STREAMING = true;
const USE_STREAMING = false;

// Simple tool function
async function getCurrentTime() {
    return { time: new Date().toISOString() };
}

// Define the tool
const tools = [
    {
        type: 'function',
        function: {
            name: 'getCurrentTime',
            description: 'Get the current time',
            parameters: {
                type: 'object',
                properties: {},
            },
            function: getCurrentTime,
            parse: JSON.parse,
        },
    },
];

async function main() {
    const stream = await openai.beta.chat.completions.runTools({
        model: 'groq/llama3-groq-70b-8192-tool-use-preview',
        messages: [
            { role: 'system', content: 'Use your tools smartly' },
            { role: 'user', content: 'What time is it now? Use your tool' },
        ],
        tools: tools,
        stream: USE_STREAMING,
    })
        .on('functionCall', (functionCall) => console.log('functionCall', functionCall))
        .on('functionCallResult', (functionCallResult) => console.log('functionCallResult', functionCallResult))
    if (USE_STREAMING) {
        for await (const chunk of stream) {
            if (chunk.choices[0].delta.content) {
                console.log(chunk.choices[0].delta);
            }
        }
    }
    const completion = await stream.finalChatCompletion();
    console.log(JSON.stringify(completion, null, 2));
}

main().catch(console.error);
```


## Config 

```yaml
model_list:
  - model_name: "*"             
    litellm_params:
      model: "*"
```

## Startup command

```
litellm --config /path/to/config.yaml 
```
